app/src/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -217,21 +217,24 @@ public static Bitmap getBitmapScaledToDisplay(File f, int screenHeight, int scre
             scale = 1;
         }
 
-        return performSafeScaleDown(f, scale);
+        return performSafeScaleDown(f, scale, 0);
     }
 
     /**
      * Returns a scaled-down bitmap for the given image file, progressively increasing the
      * scale-down factor by 1 until allocating memory for the bitmap does not cause an OOM error
      */
-    private static Bitmap performSafeScaleDown(File f, int scale) {
+    private static Bitmap performSafeScaleDown(File f, int scale, int depth) {
+        if (depth == 8) {
+            // Limit the number of recursive calls
+            return null;
+        }
         BitmapFactory.Options options = new BitmapFactory.Options();
         options.inSampleSize = scale;
         try {
             return BitmapFactory.decodeFile(f.getAbsolutePath(), options);
         } catch (OutOfMemoryError e) {
-            scale++;
-            return performSafeScaleDown(f, scale);
+            return performSafeScaleDown(f, ++scale, ++depth);
```

Comment:
nitpick: try to avoid using `var++` if you aren't then going to use the var being updated in multiple places. In this case `var + 1` suffices and tells the reader that you don't care about `var` being incremented anywhere else.

Also, I always get confused by the difference of `var++` and `++var`. The semantics of `var + 1` don't make me have to think at all :)